### PR TITLE
feat(e2e): add a job to run e2e on linux os 

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
- Add `Run Desktop E2E test` label to trigger e2e jobs on Mac OS, linux and windows
- [Doc](https://docs.google.com/document/d/1TwoHRqKXzXQ8x8zMmxrPA5ILJfjXDesM_PA5yqO7NzU/edit?usp=sharing) on how to run

```release-note
NONE
```

### CODE REVIEW

**Review by devinbinnie — APPROVED** [review]
Approving to unblock, thanks again @yasserfaraazkhan for taking this on :)

**Comment by yasserfaraazkhan** [comment]
https://community.mattermost.com/core/pl/me9rbky3mbd85qwmqhj1nkebbr

example report

File: .github/workflows/e2e-functional.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        run: |
+          wget -qO - https://download.opensuse.org/repositories/Emulators:/Wine:/Debian/xUbuntu_22.04/Release.key | sudo apt-key add -
+          wget -qO /usr/local/bin/yq https://github.com/mikefarah/yq/releases/download/v4.20.1/yq_linux_amd64 && chmod a+x /usr/local/bin/yq
+          sudo apt-get update || true && sudo apt-get install -y ca-certificates libxtst-dev libpng++-dev gcc-aarch64-linux-gnu g++-aarch64-linux-gnu jq icnsutils graphicsmagick tzdata
+          npm ci
+
+      - name: Set Environment Variables
+        run: |
+          if [ "${{ github.event_name }}" == "pull_request" ]; then
+            echo "BRANCH=${{ github.event.pull_request.title }}" >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (yasserfaraazkhan) [reply .github/workflows/e2e-functional.yml:48]: Sure @saturninoabril , will explore options for this

File: .github/workflows/e2e-functional.yml [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+  TEST_CYCLE_LINK_PREFIX: ${{ secrets.MM_DESKTOP_E2E_TEST_CYCLE_LINK_PREFIX }}
+  AWS_ACCESS_KEY_ID: ${{ secrets.MM_DESKTOP_E2E_AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.MM_DESKTOP_E2E_AWS_SECRET_ACCESS_KEY }}
+  WEBHOOK_URL: ${{ secrets.MM_DESKTOP_E2E_WEBHOOK_URL }}
+  ZEPHYR_API_KEY: ${{ secrets.MM_DESKTOP_E2E_ZEPHYR_API_KEY }}
+
+
+jobs:
+  e2e-linux:
+    if: ${{ github.event.label.name == 'Run Desktop E2E Tests' }}
```
Comment (toninis) [inline .github/workflows/e2e-functional.yml:31]: ❤️

**Review by hmhealey — APPROVED** [review]

**Pull request merged**